### PR TITLE
feat: initial scaffolding and development

### PR DESCRIPTION
**Pull request by bryce-mcmath**
Realized I still need to add GHA and Husky hooks, but hoping I could get my Helm charts looked over in the mean time. I also tried adding a devcontainer but it doesn't want to work for some reason. Doesn't give any error message, just quits when I try to start it. Not sure what I'm doing wrong yet.

Edit: GitHub Actions added. Decided not to add Husky hooks for this project and is it fairly simple. Also removed devcontainer

### CODE REVIEW

File: charts/server/templates/service.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: {{include "indy-vdr-proxy-server.fullname" .}}
+  labels: 
+    app.kubernetes.io/component: server
+    {{- include "indy-vdr-proxy-server.labels" . | nindent 4}}
+spec:
+  ports:
+    - port: {{.Values.service.targetPort}}
+      targetPort: {{.Values.service.targetPort}}
+      protocol: TCP
```
Comment (jleach) [inline charts/server/templates/service.yaml:13]: I think `TCP` is default and may not be needed.

File: charts/server/values_prod.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+podAnnotations: {}
+podLabels: {}
+
+image:
+  pullPolicy: IfNotPresent
+  registry: artifacts.developer.gov.bc.ca/github-docker-remote
+  # registry: ghcr.io
+  repository: bcgov/indy-vdr-proxy-server/server
+  # Overrides the image tag whose default is the chart appVersion.
+  # tag: "68a3892"
```
Comment (jleach) [inline charts/server/values_prod.yaml:20]: `latest`?

File: charts/server/values_test.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+podAnnotations: {}
+podLabels: {}
+
+image:
+  pullPolicy: IfNotPresent
+  registry: artifacts.developer.gov.bc.ca/github-docker-remote
+  # registry: ghcr.io
+  repository: bcgov/indy-vdr-proxy-server/server
+  # Overrides the image tag whose default is the chart appVersion.
+  # tag: "68a3892"
```
Comment (jleach) [inline charts/server/values_test.yaml:20]: `latest`?

File: src/app.module.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import { AppController } from './app.controller'
+import { AppService } from './app.service'
+import { ConsoleLogger } from '@credo-ts/core'
+import { IndyVdrPoolConfig } from '@credo-ts/indy-vdr'
+import { Logger } from '@nestjs/common'
+import { IndyVdrProxyModule, setupAgent } from 'credo-ts-indy-vdr-proxy-server'
+import { readFileSync } from 'fs'
+import 'dotenv/config'
+
+const configPath = process.env.INDY_VDR_PROXY_CONFIG_PATH ?? './res/app.config.json'
```
Comment (jleach) [inline src/app.module.ts:13]: Ok for now but we should put this into a config module named config so we can `config.get("blarb")` and supply default values.

**Review by jleach — CHANGES_REQUESTED** [review]

File: charts/server/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          ports:
+            - containerPort: {{ .Values.env.PORT }}
+              name: http
+          livenessProbe:
+            httpGet:
+              path: /health
```
[the remark this answers is not in the document]
Comment (bryce-mcmath) [reply charts/server/templates/deployment.yaml:38]: yes

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,38 @@
+FROM nikolaik/python-nodejs:python3.9-nodejs18
```
[the remark this answers is not in the document]
Comment (bryce-mcmath) [reply Dockerfile]: done

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+# RUN apk update && apk add --no-cache libc6-compat
+
+# Install and use yarn 4.x
+RUN corepack enable && corepack prepare yarn@${YARN_VERSION}
+
+# Create app directory
+WORKDIR /app
+
+# Create non-root user for Docker
+RUN addgroup --system --gid 1001 local_node
```
[the remark this answers is not in the document]
Comment (bryce-mcmath) [reply Dockerfile]: done

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,38 @@
+FROM nikolaik/python-nodejs:python3.9-nodejs18
+
+# Setup env variable for yarn
+ENV YARN_VERSION=4.3.1
+
+# Update dependencies, add libc6-compat and python to the base image
+# RUN apk update && apk add --no-cache libc6-compat
+
+# Install and use yarn 4.x
+RUN corepack enable && corepack prepare yarn@${YARN_VERSION}
```
[the remark this answers is not in the document]
Comment (bryce-mcmath) [reply Dockerfile:11]: done

File: charts/server/Chart.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v2
+name: indy-vdr-proxy-server
```
[the remark this answers is not in the document]
Comment (bryce-mcmath) [reply charts/server/Chart.yaml]: done

File: charts/server/values_dev.yaml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+podAnnotations: {}
+podLabels: {}
+
+image:
+  pullPolicy: Always
+  registry: artifacts.developer.gov.bc.ca/github-docker-remote
+  # registry: ghcr.io
+  repository: bcgov/indy-vdr-proxy-server/server
+  # Overrides the image tag whose default is the chart appVersion.
+  # tag: "68a3892"
```
[the remark this answers is not in the document]
Comment (bryce-mcmath) [reply charts/server/values_dev.yaml]: done

File: charts/server/values_dev.yaml [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+autoscaling:
+  enabled: false
+
+podAnnotations: {}
+podLabels: {}
+
+image:
+  pullPolicy: Always
+  registry: artifacts.developer.gov.bc.ca/github-docker-remote
```
[the remark this answers is not in the document]
Comment (bryce-mcmath) [reply charts/server/values_dev.yaml:13]: done

File: charts/server/templates/service.yaml (hunk 1 above)
Comment (bryce-mcmath) [reply charts/server/templates/service.yaml]: done

File: charts/server/values_prod.yaml (hunk 2 above)
Comment (bryce-mcmath) [reply charts/server/values_prod.yaml]: done

File: charts/server/values_test.yaml (hunk 3 above)
Comment (bryce-mcmath) [reply charts/server/values_test.yaml]: done

File: .github/workflows/main.yaml [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    branches: [main]
+
+jobs:
+  lint:
+    runs-on: ubuntu-20.04
+    name: Linting
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v4
```
Comment (jleach) [inline .github/workflows/main.yaml:17]: Needed?

File: .github/workflows/main.yaml [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+        run: |
+          yarn lint:check
+
+  test:
+    runs-on: ubuntu-20.04
+    name: Testing
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v4
```
Comment (jleach) [inline .github/workflows/main.yaml:45]: needed?

**Review by jleach — DISMISSED** [review]
Two comments but not blockers. Wondering if Python setup is needed in main.yaml.

File: .github/workflows/main.yaml (hunk 12 above)
Comment (bryce-mcmath) [reply .github/workflows/main.yaml:17]: Yes, a specific version of Python is needed for `ffi-napi` and `ref-napi` build process, also wanted to match the Dockerfile version

File: .github/workflows/main.yaml (hunk 13 above)
Comment (bryce-mcmath) [reply .github/workflows/main.yaml:45]: Yes, a specific version of Python is needed for `ffi-napi` and `ref-napi` build process, also wanted to match the Dockerfile version

**Review by jleach — APPROVED** [review]

**Pull request merged**